applicationset/controllers/applicationset_controller.go

```diff
@@ -133,9 +134,40 @@ func (r *ApplicationSetReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{RequeueAfter: ReconcileRequeueOnValidationError}, nil
 	}
 
+	applications, err := r.getCurrentApplications(ctx, applicationSetInfo)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	_, err = r.updateApplicationSetApplicationStatus(ctx, &applicationSetInfo, applications)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	appDependencyList, appStepMap, err := r.buildAppDependencyList(ctx, applicationSetInfo, desiredApplications)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	appSyncMap, err := r.buildAppSyncMap(ctx, applicationSetInfo, appDependencyList)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	_, err = r.updateApplicationSetApplicationStatusProgress(ctx, &applicationSetInfo, appSyncMap, appStepMap)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
 	var validApps []argov1alpha1.Application
 	for i := range desiredApplications {
 		if validateErrors[i] == nil {
+			// check appSyncMap to determine which Applications are ready to be updated and which should be skipped
+			strategy := applicationSetInfo.Spec.Strategy
+			if strategy != nil && strategy.Type == "RollingUpdate" && !appSyncMap[desiredApplications[i].Name] {
+				log.Infof("skipping application generation, waiting for previous rollouts to complete before updating: %v", desiredApplications[i].Name)
+				continue
+			}
```

Comment:
This logic works, but is this an acceptable way to prevent updates to a particular Application resource?

Reply:
I think it is not, may be create another new slice for rolligupdateApps

Reply:
Updated to run a separate filter loop before validating the desired Application resources.